react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+export class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      openIndex: -1
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(index) {
+    if (this.state.openIndex === index) {
+      this.setState({
+        openIndex: -1
+      });
+    } else {
+      this.setState({
+        openIndex: index
+      });
+    }
+  }
+
+  render() {
+    return (
+      <>
+        <button className="accordion" onClick={() => this.handleClick(0)}>Section 1</button>
```

Comment:
while this works, you don't want to hard code the index value, you want to map over the data as an array and create the buttons. That will allow you to use the `index` parameter of a map function call

Reply:
Refactored the code so that each element is dynamically rendered